compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1696,7 +1696,8 @@ abstract class GenJSCode extends plugins.PluginComponent
           else            genExpr(tree)
       }
 
-      if (!isScalaJSDefinedJSClass(currentClassSym)) {
+      if (!isScalaJSDefinedJSClass(currentClassSym) ||
+          isRawJSFunctionDef(currentClassSym)) {
```

Comment:
There are uglier things surrounding anonymous functions in the scalac backend. It doesn't bother me that much.